actors/builtin/reward/reward_test.go

```diff
@@ -1 +1,191 @@
+
+	t.Run("block reward with no payable reward", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		miner := tutil.NewIDAddr(t, 1000)
+		penalty := abi.NewTokenAmount(10)
+		gasreward := abi.NewTokenAmount(10)
+		nominalpwr := abi.NewStoragePower(10)
+		rt.SetBalance(gasreward)
+		actor.blockRewardAndVerify(rt, miner, penalty, gasreward, nominalpwr)
+
+		var st reward.State
+		rt.GetState(&st)
+		rewardMap := adt.AsMultimap(adt.AsStore(rt), st.RewardMap)
+		assert.Equal(t, big.Zero(), st.RewardTotal)
+		var r []reward.Reward
+		var rwd reward.Reward
+		err := rewardMap.ForEach(reward.AddrKey(miner), &rwd, func(i int64) error {
+			r = append(r, rwd)
+			return nil
+		})
+		require.NoError(t, err)
+		// there are 0 rewards in the actors state
+		assert.Equal(t, 0, len(r))
+	})
+
+	t.Run("block reward with reward payable", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		miner := tutil.NewIDAddr(t, 1000)
+		penalty := abi.NewTokenAmount(10)
+		gasreward := abi.NewTokenAmount(10)
+		nominalpwr := abi.NewStoragePower(10)
+		rt.SetBalance(big.NewInt(20))
+		actor.blockRewardAndVerify(rt, miner, penalty, gasreward, nominalpwr)
+
+		var st reward.State
+		rt.GetState(&st)
+		rewardMap := adt.AsMultimap(adt.AsStore(rt), st.RewardMap)
+		assert.Equal(t, big.NewInt(10), st.RewardTotal)
+		var r []reward.Reward
+		var rwd reward.Reward
+		err := rewardMap.ForEach(reward.AddrKey(miner), &rwd, func(i int64) error {
+			r = append(r, rwd)
+			return nil
+		})
+		require.NoError(t, err)
+		// there is a single reward in the actor state.
+		require.Equal(t, 1, len(r))
+
+		assert.Equal(t, abi.ChainEpoch(0), r[0].StartEpoch)
+		assert.Equal(t, 0+reward.RewardVestingPeriod, r[0].EndEpoch)
+		assert.Equal(t, big.NewInt(10), r[0].Value)
+		assert.Equal(t, big.Zero(), r[0].AmountWithdrawn)
+		assert.Equal(t, reward.RewardVestingFunction, r[0].VestingFunction)
+
+	})
+
+	t.Run("block reward send failure", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		miner := tutil.NewIDAddr(t, 1000)
+		penalty := abi.NewTokenAmount(10)
+		gasreward := abi.NewTokenAmount(10)
+		nominalpwr := abi.NewStoragePower(10)
+		rt.SetBalance(gasreward)
+		rt.ExpectValidateCallerAddr(builtin.SystemActorAddr)
+		rt.ExpectSend(builtin.BurntFundsActorAddr, builtin.MethodSend, nil, penalty, nil, exitcode.ErrForbidden)
+		rt.ExpectAbort(exitcode.ErrForbidden, func() {
+			rt.Call(actor.AwardBlockReward, &reward.AwardBlockRewardParams{
+				Miner:        miner,
+				Penalty:      penalty,
+				GasReward:    gasreward,
+				NominalPower: nominalpwr,
+			})
+		})
+		rt.Verify()
+	})
+
+	t.Run("assertion failure when miner is not ID-address protocol", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		miner := tutil.NewSECP256K1Addr(t, "bleepbloob")
+		penalty := abi.NewTokenAmount(10)
```

Comment:
If not semantically meaningful, use zero. Even better, inline big.Zero where this used.